tests/h/streamer/tweens_test.py

```diff
@@ -0,0 +1,43 @@
+from unittest import mock
+
+import pytest
+
+from h.streamer import close_db_session_tween_factory
+
+
+class TestCloseTheDBSessionTweenFactory:
+    def test_it_calls_the_handler(
+        self, close_db_session_tween, handler, pyramid_request
+    ):
+        close_db_session_tween(pyramid_request)
+
+        handler.assert_called_once_with(pyramid_request)
+
+    def test_it_closes_the_db_session(self, close_db_session_tween, pyramid_request):
+        close_db_session_tween(pyramid_request)
+
+        pyramid_request.db.close.assert_called_once_with()
+
+    def test_it_closes_the_db_session_if_an_exception_is_raised(
+        self, close_db_session_tween, handler, pyramid_request
+    ):
+        handler.side_effect = RuntimeError("test_error")
+
+        with pytest.raises(RuntimeError, match="^test_error$"):
+            close_db_session_tween(pyramid_request)
+
+        pyramid_request.db.close.assert_called_once_with()
+
+    @pytest.fixture
+    def close_db_session_tween(self, handler):
+        return close_db_session_tween_factory(handler, mock.sentinel.registry)
+
+    @pytest.fixture
+    def handler(self):
+        handler = mock.create_autospec(lambda request: None)  # pragma: nocover
+        return handler
```

Comment:
Two separate lines are used so that if the handler fixture as a whole were unused that would still show up in the coverage report

Reply:
Can this be achieved with?

```
return blah(
    lambda ... #pragma: nocover
)
```

Reply:
Sadly Black will not allow this